common/extended_window.hpp

```diff
@@ -4,6 +4,7 @@
 #include <X11/Xlib.h>
 #include <GL/glx.h>
 #include <GL/glu.h>
+#include "phonebook.hpp"
```

Comment:
Why is this include required?

Reply:
`phonebook::service` on line 15.

Suppose some file `A` includes this file, `extended_window.hpp`. `A` included `phonebook.hpp` for unrelated reasons, prior to including `extended_window.hpp`. That is why it was not causing an error.